agent/consul/server.go

```diff
@@ -1170,20 +1170,32 @@ func (s *Server) Shutdown() error {
 	return nil
 }
 
-func (s *Server) attemptLeadershipTransfer() (success bool) {
+func (s *Server) attemptLeadershipTransfer(id raft.ServerID) (err error) {
 	leadershipTransferVersion := version.Must(version.NewVersion(LeaderTransferMinVersion))
 
 	ok, _ := ServersInDCMeetMinimumVersion(s, s.config.Datacenter, leadershipTransferVersion)
 	if !ok {
-		return false
+		return fmt.Errorf("minimum version not supported, minimu required version %s", LeaderTransferMinVersion)
 	}
 
```

Comment:
This whole block can be deleted. Everything at least as far back as 1.9 will support this and we will not support <1.9 to 1.14 upgrades directly without passing through 1.9, 1.10, 1.11, 1.12, or 1.13